ironfish/src/primitives/rawTransaction.ts

```diff
@@ -54,17 +58,54 @@ export class RawTransaction {
     >
   }[] = []
 
-  size(): number {
+  postedSize(publicAddress: string): number {
     let size = 0
+    size += 1 // version
     size += 8 // spends length
     size += 8 // notes length
-    size += 8 // fee
-    size += 4 // expiration
-    size += 64 // signature
+    size += 8 // mints length
+    size += 8 // burns length
+    size += TRANSACTION_FEE_LENGTH // fee
+    size += TRANSACTION_EXPIRATION_LENGTH // expiration
+    size += TRANSACTION_PUBLIC_KEY_RANDOMNESS_LENGTH // public key randomness
+    size += this.spends.length * SPEND_SERIALIZED_SIZE_IN_BYTE
     size += this.outputs.length * NOTE_ENCRYPTED_SERIALIZED_SIZE_IN_BYTE
-    size += this.mints.length * (ASSET_LENGTH + 8)
+    size +=
+      this.mints.length *
+      (PROOF_LENGTH + ASSET_LENGTH + AMOUNT_VALUE_LENGTH + TRANSACTION_SIGNATURE_LENGTH)
     size += this.burns.length * (ASSET_ID_LENGTH + 8)
-    size += this.spends.length * SPEND_SERIALIZED_SIZE_IN_BYTE
+    size += TRANSACTION_SIGNATURE_LENGTH // signature
+
+    // Each asset might have a change note, which would need to be accounted for
+    const assetTotals = new AssetBalances()
+    for (const mint of this.mints) {
+      const asset = new Asset(publicAddress, mint.name, mint.metadata)
+      const assetValue = assetTotals.get(asset.id())
+      assetTotals.set(asset.id(), assetValue ? assetValue + mint.value : mint.value)
```

Comment:
Should we use `assetTotals.increment()` here?

Reply:
Oh yea that is real nice! Cleans up this code a bunch. TY